Fleece/MSVC_Compat.hh

```diff
@@ -38,6 +38,10 @@
     // WARNING: sizeof() will not work on this array since it's actually declared as a pointer.
     #define StackArray(NAME, TYPE, SIZE)    TYPE* NAME = (TYPE*)_malloca(sizeof(TYPE)*(SIZE))
 
+    #define strlcat strncat
```

Comment:
I figured out that part but wouldn't a null in the middle of a non terminated multi byte character also be corrupt later when attempting to read it as utf-8?  It probably wouldn't be as bad but it's something that worries me. That's likely to happen if the buffer can't quite fit the entire sequence and then it simply ensures the last character is null.  Is that situation recoverable?  Note that this isn't an argument in favor of strncpy, just a concern overall.